src/elements/content-sidebar/activity-feed/task/Task.js

```diff
@@ -72,6 +73,18 @@ class Task extends React.Component<Props> {
             can_delete: isCommentAuthor,
         };
 
+        // // find assignment for current user if permitted
+
+        const currentUserAssignment =
+            task_assignment_collection && task_assignment_collection.entries
```

Comment:
You could clean this up a little via something like `const taskAssignments = getProp(task_assignment_collection, 'entries', [])` and then map/filter over that?

Reply:
Yeah something like that. Working on extracting, memoizing, etc. for the computations once all the computed entities are there. Still need some more.